docs/registry/README.md

```diff
@@ -0,0 +1,193 @@
+# Registry Proposal
+
+## Objective
+
+Design an **initial** version of the Registry that can support discoverability of 
+the different event types that can be consumed from the eventing mesh.
```

Comment:
I think it would be good to add a bit more here w.r.t. to justification. For example, I believe that a large number of usecases will be of the form:
- user wants to get events from event producer X
- user looks at X's docs, or something, to see what event types they can subscribe to
- user creates an X eventSource specifying which event types they want to subscribe to from X - or specify "all"

In this flow, it's not clear to me why a user would need a registry. Why would they not always just go to the source of truth w.r.t. what X will produce. What value does the registry provide that going to X doesn't?  I think many people will have these types of questions so adding text here will really help explain why this is needed.

And related to this, some event sources actually change the eventType value generated by X. I worry that a registry will encourage more of this and that's a concern because I think it adds a layer of mapping/indirection for users that will add work/confusion.  As an app author, I'd like my code to work in as many platforms as possible, with minimal code changes. If the code I write would normally look for a type of "foo" but now I have to look for "KnFoo" when running in Kn, that's an additional pain-point for people. Yes, if there is no clear value from X to use then Kn needs to create something, but in cases like Github I know there are values already and yet we do a mapping.

Reply:
IMHO one of the main values of the registry is to provide an *inbound* mechanism for users to see what they can consume. Users won't need to go OOB to figure out what those events are.
Also for admission control, e.g., if it's not in the registry, don't let that event in, and so forth.

Regarding the eventType thing, I'm (mostly) with you on this one. I think we shouldn't change the type. If we want to distinguish that a github pull_request came from our adapters instead of straight from github (whenever they start sending cloud events), then let's just add a cloud event custom extension. But I know that @n3wscott has a good justification of why we are currently changing them.

Will make another pass on the README soon to try to address your concerns. Thanks for the comment @duglin!

Reply:
One of the benefits for the user IMHO with a knative based platform is the ability to mix and "mesh in" different event sources and do events based integration. For such a case of multiple heterogeneous sources and events coming into the cluster, we want the platform to play "for" the user and help the user find/discover and choose the correct event-type.

 This is not just a "fancy tooltip" style of enhancement, this is IMHO a core  aspect of the user experience that might end up being a key differentiator feature. I know for sure that the majority of our customers heavily endorse that inside Kyma you can find out 
- What are the events types an external application (event source) might send.
- How does the external application API look like. 
- How does the configuration for a service from a cloud provider service catalog look like.

And being able to reach such info without the need to go OOB is the trick here.

I think we can discuss the pros and cons and what are different design alternatives, however IMHO this is a feature that will benefit a huge no of users and will increase the adoption of platform built on top of knative.